### DISCUSSION TITLE
Images as a choice

### DISCUSSION THREAD

**Discussion by gwwharton** [inkle/ink#851] [General]
Is it possible to make an image as a choice that diverts? Similarly to what is available now with simple text?

**Comment by Selsynn** [comment]
You can do it, but you'll need to tinker in the view itself. It depends what you are using. If using a web export, you can just put the image in HTML, I think it will work, as long as it is in the button that will link it to the player action. 
(`<img src="URL of the image" alt="text for if the image doesn't display"/>` I didn't test it, but it should work out of the box.)


If you are in unity, tho, you have quite some tinkering to do with your view.
Usually what I do is to have the view transform the following ink choice 
```ink
* [@IMG@Id_ref_img@ ] Text once the choice is taken
```
to the image in the button I want. 
You just need to parse the choice text for the "@" and get the local image path. You'll want to make sure you have the correct image available too and have a  "by default" image or some kind of other way to not break your game because there is a typo in the name of the image.

**Reply by gwwharton** [reply]
Thank you for the reply. I get what your saying with the HTML but I cannot get into the choice itself since its inside the javascript

**Reply by Selsynn** [reply]
For the web in html, as long as the "logo.png" exist at the root of the export folder, it works 
```
 * [ \<img src="./logo.png"\> ] There were four lines of content.
 ````

I'm not sure why you have to escape the `<>` but it didn't want to work without.

In js you don't have to touch the js, it's just in the ink, as it is directly handed as HTML.

**Reply by gwwharton** [reply]
That worked! Thank you! So, in Ink can you write in HTML as long as you have slashes? 
Example:

`===paragraph_1===`
`Lines of content. Lines of content.  [ \<htmlhtml\> ] Lines of content.`

**Reply by Selsynn** [reply]
I suggest that you make sure that any HTML code you want to test works on a smaller project, so that you could just list every block you want in the choice or in the text. You export this one as a web project, and look what is working as you expect and what is not, so you know before putting it into your "big" project.

But yeah, I think it should be good. As I write earlier, I'm not sure why it needs to escape the characters. Maybe inky is thinking it's the glue? Not sure.
